src/consensus/params.h

```diff
@@ -11,13 +11,25 @@
 
 namespace Consensus {
 
-enum DeploymentPos
+enum BuriedDeployment : int16_t
+{
+    // buried deployments get negative values to avoid overlap with SignalledDeployment
+    DEPLOYMENT_HEIGHTINCB = std::numeric_limits<int16_t>::min(),
+    DEPLOYMENT_CLTV,
+    DEPLOYMENT_DERSIG,
+    DEPLOYMENT_CSV,
+    DEPLOYMENT_SEGWIT,
+};
+constexpr bool ValidDeployment(BuriedDeployment dep) { return DEPLOYMENT_HEIGHTINCB <= dep && dep <= DEPLOYMENT_SEGWIT; }
+
+enum DeploymentPos : uint16_t
 {
     DEPLOYMENT_TESTDUMMY,
     DEPLOYMENT_TAPROOT, // Deployment of Schnorr/Taproot (BIPs 340-342)
     // NOTE: Also add new deployments to VersionBitsDeploymentInfo in versionbits.cpp
```

Comment:
I think this comment was previously incorrect (`VersionBitsDeploymentInfo` was in `versionbitsinfo.cpp`), but should definitely be updated since as of this PR it is defined in `deploymentinfo.cpp`